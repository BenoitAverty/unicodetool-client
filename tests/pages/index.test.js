import { shallow, mount, render } from 'enzyme'
import toJson from 'enzyme-to-json'

import { withoutRedux as Index } from '../../src/pages/index.js'

describe('Index page', () => {
  it('matches snapshot', () => {
    const page = shallow(
      <Index currentSearch='current search' handleSearchChange={() => {}} />
    )

    expect(toJson(page)).toMatchSnapshot()
  })

  it('fires the proper callback when the search field changes', () => {
    const callbackMock = jest.fn()
    const page = shallow(
      <Index currentSearch='current search' handleSearchChange={callbackMock} />
    )

    page.find('UnicodeSearchField').simulate('change', 'new value')

    expect(callbackMock.mock.calls.length).toBe(1)
    expect(callbackMock.mock.calls[0][0]).toBe('new value')
  })
})
