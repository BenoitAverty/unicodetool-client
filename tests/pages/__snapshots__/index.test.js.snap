// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page matches snapshot 1`] = `
<Layout>
  <jf
    tag="div"
  >
    <UnicodeSearchField
      onChange={[Function]}
      value="current search"
    />
    <p
      data-jsx={475681380}
    >
      Examples: "
      <a
        data-jsx={475681380}
        href="#"
      >
        U+0041
      </a>
      ", "
      <a
        data-jsx={475681380}
        href="#"
      >
        GHOST
      </a>
      ", "
      <a
        data-jsx={475681380}
        href="#"
      >
        Ã 
      </a>
      ".
       
      <Link
        href="/help"
      >
        <a
          data-jsx={475681380}
        >
          More help and examples...
        </a>
      </Link>
    </p>
  </jf>
  <_class
    css="p[data-jsx=\\"475681380\\"] {text-align: center;}"
    styleId={475681380}
  />
</Layout>
`;
